### ISSUE TITLE
Rule S1172: false positive with nameof

### ISSUE THREAD

**Issue by shuebner** [SonarSource/sonar-csharp#1194]
### Description

S1172 gives a false positive on a method parameter which is only used in a nameof expression in a private method.
I guess this is awkward and an edge case at best.

### Repro steps

```csharp
public class Class1
{
    public string Bar()
    {
        return this.Foo("baz");
    }

    private string Foo(string baz)
    {
        return $"{nameof(baz)}";
    }
}
```

### Expected behavior

Not sure because it is an awkward edge case.

### Actual behavior

S1172 complains about `baz` being unused.

### Related information

* SonarC# Version 6.8.1.4648
* Visual Studio Version 15.5.0

**Comment by michalb-sonar** [comment]
This is indeed an odd case @shuebner. Note this code only uses the name of the variable - so it's equivalent to `return "baz";`
The value of the parameter is unused - whatever you pass as `baz`, the behaviour won't change - so I think the rule was right to report, as it's suspicious.
If you see any cases when it would be an useful pattern, we'd be happy to reconsider.

**Comment by Evangelink** [comment]
@shuebner,

I am guessing you just created a small example to reproduce the issue. Would you mind sharing a more complex example (or the actual code)?

**Comment by shuebner** [comment]
The discovery was the result of some mistyping on my part, so there is no real-life code for this.
IMHO @michalb-sonar is right on this one.

**Comment by Evangelink** [comment]
Perfect!
